contactsummary/js/src/contact-summary/controllers/key-dates.controller.js

```diff
@@ -46,47 +48,92 @@ define([
+      $log.debug('Subcsribers initialiazed!');
 
-      Contract.get()
-        .then(function (response) {
-          if (!_.isEmpty(response)) {
-            angular.forEach(response, function (contract) {
-              addContractDates(contract);
+      var events = [
+        'Contract::created',
+        'Contract::updated',
+        'JobRole::created',
+        'JobRole::updated',
+        'JobRole::deleted'
+      ];
 
-              if (contract.is_current === '1') {
-                vm.activeContracts++;
-              }
-            });
-          }
+      events.forEach(function (event) {
+        pubSub.subscribe(event, loadData);
+      });
 
-          return JobRole.get();
-        })
+      pubSub.subscribe('Contract::deleted', function (contract) {
+        Contract.removeContract(contract);
+        loadData();
+      });
+    }
+
+    /**
+     * Checks if a date is in the future
+     *
+     * @param  {String} date
+     * @return {Boolean}
+     */
+    function isDateNotInPast (date) {
+      return moment().diff(date, 'days') <= 0;
+    }
+
+    /**
+     * Loads contracts from the backend and sets their amount to the controller
+     *
+     * @return {Promise}
+     */
+    function loadContactsAmount () {
+      return Contract.get()
         .then(function (response) {
-          angular.forEach(response, function (role) {
-            var endDate = moment(role.end_date);
+          vm.activeContracts = _.values(response).filter(function (contract) {
+            addContractDates(contract);
 
-            if (role.end_date === undefined || (endDate.isValid() && isDateNotInPast(endDate))) {
-              vm.activeRoles++;
-            }
-          });
-        })
+            return contract.is_current;
+          }).length;
+        });
+    }
+
+    /**
+     * Fetches contacts and job roles from the backend
+     *
+     * @return {Promise}
+     */
+    function loadData () {
+      vm.ready = false;
+
+      resetKeyDates();
+
+      return $q
+        .all([
+          loadContactsAmount(),
+          loadJobRolesAmount()
+        ])
         .finally(function () {
           vm.ready = true;
         });
     }
 
     /**
-     * Checks if a date is in the future
+     * Loads job roles from the backend and sets their amount to the controller
      *
-     * @param {string} date
-     * @return {boolean}
+     * @return {Promise}
      */
-    function isDateNotInPast (date) {
-      return moment().diff(date, 'days') <= 0;
+    function loadJobRolesAmount () {
+      return JobRole.get()
+        .then(function (response) {
+          vm.activeRoles = response.filter(function (role) {
+            var endDate = moment(role.end_date);
+            var hasEndDateAndIsActive = endDate.isValid() && isDateNotInPast(endDate);
+            var doesNotHaveEndDate = endDate === undefined;
+
+            return doesNotHaveEndDate || hasEndDateAndIsActive;
+          }).length;
+        })
+        .catch(_.noop);
```

Comment:
tbh loading active job roles is something I rather have in the model. If you can spare a few minutes to move this to the model it would be great!

Reply:
Do you mean the `reqangular` JobRoles model?